### PR TITLE
[Fixes #10142] storage_manager copy dont assign the folder/file permi…

### LINKED ISSUE #10142: storage_manager copy dont assign the folder/file permissions correcly
## Expected Behavior
Storage manager during the copy should set the right permission at the file and folder level

## Actual Behavior
The permissions are not assigned. 

## Specifications

  - GeoNode version: master/4.0.x
  - Installation type (vanilla, geonode-project): all
  - Installation method (manual, docker): all
  - Platform: all
  - Additional details: is related to the work started with this issue -> https://github.com/GeoNode/geonode/issues/10130

### PR DESCRIPTION
**Pull request by mattiagiupponi**
…ssions correctly
ref #10142 
<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/storage/manager.py [hunk 1]
```diff
@@ -199,7 +205,9 @@ def copy_files_list(self, files: List[str]):
                     new_file = f"{new_path}/{suffixed_name}{ext}"
                 else:
                     new_file = f"{new_path}/{old_file_name}_{random_suffix}{ext}"
-                out.append(self.copy_single_file(open_file, new_file))
+                _new_path = self.copy_single_file(open_file, new_file)
+                out.append(_new_path)
+            os.chmod(_new_path, settings.FILE_UPLOAD_PERMISSIONS)
```
Comment (afabiani) [inline geonode/storage/manager.py:210]: I guess there's an indentation error here

**Review by afabiani — CHANGES_REQUESTED** [review]

File: geonode/storage/manager.py (hunk 1 above)
Comment (mattiagiupponi) [reply geonode/storage/manager.py:210]: the cmod should be done once the file is closed before taking the next item in the loop

Comment (afabiani) [reply geonode/storage/manager.py:210]: but the `_new_path` variable is not defined there I guess, it's inside the `with` statement

Comment (mattiagiupponi) [reply geonode/storage/manager.py]: i add and additional check to verify that _new_path is filled before re-assign the permission. Is wanted to be done just before the loop so we can ensure that the file is closed before trying to access it

**Review by afabiani — APPROVED** [review]

**Pull request merged**